app/helpers/katello/katello_urls_helper.rb

```diff
@@ -37,11 +37,36 @@ def subscription_manager_configuration_url(host = nil, rpm = true, hostname: nil
       "#{prefix}/pub/#{config}"
     end
 
-    apipie :method, 'Generates an absolute path to the file' do
+    apipie :method, 'Generates an absolute path to the file for liveimg kickstart templates' do
       required :content_path, String, desc: "Relative path to the file or it's name"
       optional :schema, String, desc: 'Optional URL schema for the content source', default: 'http'
       optional :content_type, String, desc: 'Content type', default: 'repos'
       returns String, desc: 'Absolute path to a file'
     end
+    def repository_url(content_path, _content_type = nil, schema = 'http')
+      return content_path if content_path =~ %r|^([\w\-\+]+)://|
+      url = if @host.content_source
+              "#{schema}://#{@host.content_source.hostname}"
+            else
+              foreman_settings_url(schema)
+            end
+      content_path = content_path.sub(%r|^/|, '')
+
+      lifecycle_environment = nil
+      if @host.default_environment?
+        # Don't update the content_path if the host is using the default org view / library
+        lifecycle_environment = ::Katello::KTEnvironment.library.find_by(organization_id: @host.organization_id)
+      elsif !@host.single_content_view_environment?
```

Comment:
```suggestion
      elsif @host.multi_content_view_environment?
```

Reply:
and now that I'm here, I wonder if we should add a `!default_environment?` check to that method.. :thinking:

Reply:
I feel like there's good reason to check for single content view environment separately from checking the default environment.

Reply:
on second thought I agree.